### DISCUSSION TITLE
Is it possible to use breakpoints with cached modules?

### DISCUSSION THREAD

**Discussion by dcacenabes** [swiftyfinch/Rugby#406] [Q&A]
Hi,
We have some use cases in which our developers are using precompiled development pods but they would like to be able to have breakpoints in those, to help troubleshoot some issues.
I have been playing with updating the build config for debug builds so that it includes dsym symbols, and after running rugby build + use I am able to see the dsym files in the ~/.rugby/bin/... folder, but when I am running the app in Xcode I can't hit any breakpoint. The breakpoint shows up in Xcode surrounded by a dotted line and it displays the following message when you hover the mouse over it:
>Xcode won't pause at this breakpoint because it has not been resolved Resolving it requires that:
The line at the breakpoint is compiled.
The compiler generates debug information that is not stripped out (check the Build Settings).
The library for the breakpoint is loaded.

Is this achievable with Rugby? Maybe I am missing something very basic regarding Rugby/Xcode configuration. Any guidance will be appreciated.

**Comment by Ro-M** [comment]
Maybe this is something that has to be tried out from the Cocoapods side first, and once its working there we see how it gets properly updated with Rugby?

**Reply by dcacenabes** [reply]
With Cocoapods alone I have no problem debugging code in all the development pods. It is after running `rugby use` command that the problem appears.

**Comment by swiftyfinch** [comment]
@dcacenabes Hello! Is this topic relevant for now?

Did you compare the local build without remote binaries with the build that includes them? \
As I know, to use breakpoints, you need to have some additional files from the build folder. \
I didn't dive into this, but I'm pretty sure that the local build without remote binaries should allow breakpoints to be used.

If so, you can use it this way or find out what the files are and upload/download each one together with the common frameworks to/from your shared storage.
It would be nice to hear any news about the latest one, including an estimate of the file size)
